### PR TITLE
Fix quorum data

### PR DESCRIPTION
**Pull request by LeonVreling**
#MEMB-819

### CODE REVIEW

**Comment by LeonVreling** [comment]
Maybe someone can try to test this on their local machine, as I can't apply to statutory events

**Comment by LeonVreling** [comment]
Also @serge1peshcoff maybe you know a better location to put the calculations of the `nonQuorum` and `presentQuorum` arrays

File: src/views/statutory/Stats.vue [hunk 1]
```diff
@@ -293,12 +293,26 @@ export default {
       }
     },
     byQuorumData () {
-      const present = this.stats.by_body.length * 100 / this.bodies.length
+      const nonQuorum = []
+      for (const body of this.bodies) {
+        if (!body.name.includes('AEGEE', 0)) {
```
Comment (serge1peshcoff) [inline src/views/statutory/Stats.vue]: `if (['antenna', 'contact antenna', 'contact'].includes(body.name)) {`, this is more readable

File: src/views/statutory/Stats.vue [hunk 2]
```diff
@@ -293,12 +293,26 @@ export default {
...
+          nonQuorum.push(body)
+        }
+      }
+
+      const presentQuorum = []
+      for (const body of this.stats.by_body) {
+        if (nonQuorum.indexOf(body) === -1) {
```
Comment (serge1peshcoff) [inline src/views/statutory/Stats.vue]: Also this won't work as these are different objects and also be false

**Comment by LeonVreling** [comment]
Number 1 would not work right, as contacts are not included in the quorum, because they have not signed the CdA (I think), and also this will leave out the AEGEE-Academy, which should be included. That is why I took all the bodies with the name starting with "AEGEE-".

**Comment by serge1peshcoff** [comment]
@LeonVreling you can replace contact with partner then ;)
It's better to filter on body type and not name as, imagine if we'll have "AEGEE election observation" body which is a project but would be considered a local if we'll filter by name.

**Comment by LeonVreling** [comment]
Hm maybe with the new workflow, this should all be a 'fix', but I'll let you to decide

File: src/views/statutory/Stats.vue [hunk 3]
```diff
@@ -293,13 +293,25 @@ export default {
...
+      const localsMap = {}
+      for (const body of this.bodies) {
+        if (['antenna', 'contact antenna', 'partner'].includes(body.type)) {
+          localsMap[body.id] = true
+        }
+      }
+
+      const total = Object.keys(localsMap).length
+      const present = this.stats.by_body
+        .filter((body) => body.type in localsMap)
```
Comment (serge1peshcoff) [inline src/views/statutory/Stats.vue]: will that work? I'm not sure if it will, but if so, that's okay

**Comment by serge1peshcoff** [comment]
@LeonVreling for the new workflow, I can now squash and merge it as `fix:` and this will be okay.

File: src/views/statutory/Stats.vue (hunk 3 above)
Comment (LeonVreling) [reply src/views/statutory/Stats.vue]: Yes it works on my local device

Comment (LeonVreling) [reply src/views/statutory/Stats.vue]: When I apply as AEGEE-Eindhoven, it gets added to the quorum, but when I do so as ITC, it does not show up.

Comment (serge1peshcoff) [reply src/views/statutory/Stats.vue]: Awesome

**Comment by serge1peshcoff** [comment]
@LeonVreling I've asked Veva, she told that the body shouldn't be reflected in quorum if it didn't send any delegates/envoys, so this one is not enough. Also the stats endpoint doesn't return the required data, so it's not as simple. 
This one is anyway more correct than the prod one, so I'll merge a little bit later.

**Comment by LeonVreling** [comment]
I've made a new JIRA issue for it `#834`, so that we don't forget to update it in the future

**Pull request merged**